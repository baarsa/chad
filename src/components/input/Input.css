.title {
	margin-bottom: 8px;
}

.input {
	width: 100%;
	background: #37474f;
    height: 40px;
    line-height: 1;
    outline: 0;
    border: 0;
    border-radius: 3px;
    padding: 0 10px;
    font-size: 14px;
    color: #e0e0e0;
    box-sizing: border-box;
}