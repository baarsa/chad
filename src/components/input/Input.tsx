import * as React from 'react'
import * as styles from './Input.css'

interface Props extends React.HTMLProps<HTMLInputElement> {
	outerClass?: string;
	title: string;
	onValueChange?(value: string): void;	
}

interface State {
	value: string;
}

class Input extends React.Component<Props, State> {
	state: State = {
		value: ''
	}

	handleChange = (event: any) => {		
		const value = event.target.value;		
		this.setState(() => ({value}));
		if (this.props.onValueChange) {
			const {onValueChange} = this.props;
			onValueChange(value);			
		}		
	}

	render() {
		const { value } = this.state;
		const { outerClass, title, onValueChange, ...inputProps } = this.props;

		return (<div className={outerClass}>
			<div className={styles.title}>{title}</div>
			<input className={styles.input} {...inputProps} value={value} onChange={this.handleChange} />
		</div>)
	}
}

export default Input;