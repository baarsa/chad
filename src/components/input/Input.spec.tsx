import * as React from 'react'
import * as ReactDOM from 'react-dom'
import * as ReactTestUtils from 'react-dom/test-utils'
import {shallow, mount, render} from 'enzyme'
import {shallowToJson} from 'enzyme-to-json'
import Input from './Input'

describe("Input", () => {	
	const testTitle = "test input";

	it("should render correctly", () => {
		expect(shallow(<Input title={testTitle} />).find('div').exists()).toBe(true);
	});

	it("renders input", () => {
		expect(shallow(<Input title={testTitle} />).find('input').exists()).toBe(true);
	});
	
	it("should update state with input", () => {
		const component = shallow(<Input title={testTitle} />);
		component.find('input').simulate('change', {
			target: {
				value: 'test input'
			}
		});
		expect(component.state('value')).toEqual('test input');
	});
/*
	it("renders title somewhere", () => {
		const result = renderer.root.find(
			);		
		chai.assert.strictEqual(result.children[0].);
	});
	*/
});