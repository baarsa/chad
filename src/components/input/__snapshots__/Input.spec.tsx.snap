// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input should render correctly 1`] = `
<div>
  <div
    className="all"
  >
    test input
  </div>
  <input
    onChange={[Function]}
    value=""
  />
</div>
`;
